command/agent/agent_test.go

```diff
@@ -1851,6 +1851,74 @@ func TestAgent_purgeCheckState(t *testing.T) {
 	}
 }
 
+func TestAgent_metadata(t *testing.T) {
+	config := nextConfig()
+	dir, agent := makeAgent(t, config)
+	defer os.RemoveAll(dir)
+	defer agent.Shutdown()
+
+	// Load a valid set of key/value pairs
+	config.Meta = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+	// Should succeed
+	if err := agent.loadMetadata(config); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	agent.unloadMetadata()
+
+	// Should get error
+	config.Meta = map[string]string{
+		"": "value1",
+	}
+	if err := agent.loadMetadata(config); err == nil {
```

Comment:
It would be good in these error cases to make sure it's failing for the reason you expect (check some part of the error string)